src/main/java/org/broadinstitute/hellbender/engine/spark/ShuffleJoinReadsWithRefBases.java

```diff
@@ -51,9 +51,10 @@
  * The reference bases paired with each read can be customized by passing in a reference window function
  * inside the {@link ReferenceDataflowSource} argument to {@link #addBases}. See {@link org.broadinstitute.hellbender.engine.dataflow.datasources.RefWindowFunctions} for examples.
  */
-public class JoinReadsWithRefBases {
+public class ShuffleJoinReadsWithRefBases {
     public static JavaPairRDD<GATKRead, ReferenceBases> addBases(final ReferenceDataflowSource referenceDataflowSource,
                                                                  final JavaRDD<GATKRead> reads) {
+        // TODO: reimpl this method by calling out to the more complex version?
```

Comment:
If we had to pay the cost of an extra map at runtime then I'm not sure it would be worth it, but I defer to your judgment on this issue. I guess there's no way we could get around this by using a common ancestor class of `JavaRDD` and `JavaPairRDD` and/or passing in a function that knows how to unpack the reads from the input? I ask only because I think we're going to have this sort of problem again in the future with Spark, and it would be good to settle on a canonical solution.